src/re_com/multi_complete.cljs

```diff
@@ -0,0 +1,138 @@
+    :type "string"
+    :validate-fn string?
+    :description "class for each item in suggestion dropdown"}
+   {:name :save!
+    :type "choice -> anything"
+    :validate-fn ifn?
+    :description "a callback that will be passed either a selected item from suggestions or the value of the text in the input field"
+    }
+    {:name :on-delete!
+     :type "nil -> anything"
+     :validate-fn ifn?
+     :description "Fired whenever the text field is empty and delete button is pressed, purpose is generally to remove the last item in the model/selections atom"}
+   {:name :placeholder
+    :type "string"
+    :description "A hint to make sure users know they can type here"}]
+  )
+
+
+
+
+(defn multi-complete [{:keys [highlight-class
+                              placeholder
+                              item-class
+                              list-class
+                              suggestions
+                              save!
+                              selections
+                              on-delete]
+                       :or {highlight-class "multi-complete-highlight"
+                            item-class ""}}]
+  (let [a (r/atom "")
+        selected-index (r/atom -1)
+        typeahead-hidden? (r/atom false)
+        mouse-on-list? (r/atom false)]
+    (fn []
+      (let [options  (if (clojure.string/blank? @a)
+                       []
+                       (filter
+                        #(-> % (.toLowerCase %) (.indexOf (.toLowerCase @a)) (> -1))
+                        suggestions))
+            matching-options (filter (comp not (set @selections)) options)
+            choose-selected #(if (and (not-empty matching-options)
+                                       (> @selected-index -1))
+                              (let [choice (nth matching-options @selected-index)]
+                                (save! choice)
+                                (reset! selected-index 0)
+                                (reset! a ""))
+                              (when (not (str/blank? @a))
+                                (do
+                                  (save! @a)
+                                  (reset! selected-index 0)
+                                  (reset! a ""))))]
+        [h-box
+         :class "multi-complete"
+         :children [[:div.multi-complete-output
+                     (when @selections
+                       (for [x @selections]
+                         ^{:key x}[:button {:class "btn btn-default"
+                                          :on-click #(swap! selections (fn [y] (remove #{x} y)))}(str x)]
+                         ))
+                     [:input.multi-complete-input
+                      {:value @a
+                       :placeholder placeholder
+                       :on-change #(reset! a (-> % .-target .-value))
+                       :on-key-down #(do
+                                       (case (.-which %)
+                                         38 (do
```

Comment:
Good question, from https://google.github.io/closure-library/api/goog.events.KeyCodes.html

> This list is not localized and therefore some of the key codes are not correct for non US keyboard layouts. See comments below.

Looking at the [source](https://github.com/google/closure-library/blob/master/closure/goog/events/keycodes.js#L37) it seems like the keycodes we want don't need to be localised.